### ISSUE TITLE
Not able to save the output window

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#8408]
![image](https://github.com/africanmathsinitiative/R-Instat/assets/11226469/c2a36be2-b13a-402a-ad67-7ca8f256d8a8)

@Patowhiz what can we do for the participants on the workshop?

We are using 7.14, but I have trioed in 7.15 and this is the current merged version.

The participants want to save the output window.  Please can you help with the current version as well as sorting out a fix.  It seems to be a general problem and I am a bit at a loss that I did not notice it befoe, as I have been considering the saving aspects.

**Comment by rdstern** [comment]
@Patowhiz yes, this remains the major problem we have found, with the software so far.  They were hoping to save a record of the results through the week, each on their own data.  REal blow that they can't   Hope it will be fixed for the next workshop, which is Tajikistan in a couple of weeks.

Particular problem here, because the team used R-Instat before, so it is an item that used to work, and has now stopped.

**Comment by Patowhiz** [comment]
@ChrisMarsh82 has suggested, converting html to images as a start. I'll explore this as well.

**Comment by rdstern** [comment]
@Patowhiz and @ChrisMarsh82 I find it hard to address your options, while something this important - trhat used to work - has stopped working.  And there was no isuue on that, so I found out only during a workshop.  How did I miss the problem as I have been worrying about saving options and why didn't it surface during your visit to UK, or the sprint?

In answer to your queries:

1. I am happy for you to explore options.  I note, for example, there is the [gembox software](https://www.gemboxsoftware.com/document) that seems to do the sort of thing we require, namely to transform output into different formats.   (That is commercial, but was can consider it, if it the best option, because it is royalty free.)  

There is a free version - limited to 20 paragraphs!  Even that would be ok for now.  We need something working by the next release, which is Wednesday!  I am happy with anything, as long as there is something.

As you know I have been asking for more options for your maximised window and, in the longer term it would be good to consider the two together.  We used the maximised window a lot in the workshop, see issues #8141 and #8189.

2. Agreed - if they are disabled, then don't need to print.

3. Not needed now.  Let's concentrate on saving something.

4.  In your more general assessment, please remember the gt software permits the option of tables in different formats.  Perhaps this also relates to options for you.  

Please can you treat it as urgent to get something - however modest - working again for this upgrade (Wednesday) and then aim for a decent set of options for the August version?  For example it would be nice if <ctrl>C could work (again?) to copy to the clipboard, and if we could manage the checkboxes better, e.g mark a set of them together - and delete more than 1 at a time - issue #7979.

**Comment by Patowhiz** [comment]
@ChrisMarsh82 agreed.

@rdstern @lloyddewit I take full responsibility of the regression introduced and I've been discussing with @ChrisMarsh82 about this.

The former version worked because, it created 2 elements per dialog script. An output element that had the script produced only and another one that had the script and the output. In a nutshell, a dialog would produced 2 output elements that contained the same exact script but one doesn't have the output.
This maintained a 1:1 relationship with the output logger but contravened the conceptual intent of the output logger and output elements in regards to their relationship with the audit log.

I changed the implementation to make sure that a dialog produced one output element that contains a script and optionally an output (if the script produced an output). This fundamentally changed how saving, adding to new tab and other  functionalities should have been implemented. My changes should not have stopped there or at least not been added all together.

I did make this change to make sure that the output logger maintains a 1:1 relationship with the audit log (which is what has always been desired).

I'll do a short term fix and then we can discuss a long term fix which may include decoupling the output window and pages functionalities from the output logger while still maintaining referential relationship between the two in terms of the output elements contained.

**Comment by Patowhiz** [comment]
@rdstern thanks for suggesting the gembox software. I will look into it while implementing the long term solution.

**Comment by lilyclements** [comment]
@Patowhiz was this solved?
